### DISCUSSION TITLE
error with json_kinds at compile time

### DISCUSSION THREAD

**Discussion by jalvesz** [modern-fortran/neural-fortran#121] [Q&A]
Hi, I wanted to give a try to the library and I'm a bit stuck with a compilation problem:
I'm working on Windows with ifort19, I created a .bat file to set the environnement variables for ifort and HDF5, so far so good. when the compilation arrives to the json_kinds.F90 file I have the following warnings:

"build\dependencies\json-fortran\src\json_kinds.F90(16): warning #5117: Bad # preprocessor line"
"#ifdef USE_UCS4"
"-^"
"build\dependencies\json-fortran\src\json_kinds.F90(17): warning #5117: Bad # preprocessor line"
"# pragma push_macro("USE_UCS4")"
"--^"
...

(all lines up to 133)

Then the error:
build\dependencies\json-fortran\src\json_kinds.F90(87): error #6418: This name has already been assigned a data type.   [RK]       
    integer,parameter,public :: RK = real64   !! Default real kind [8 bytes]
--------------------------------^
build\dependencies\json-fortran\src\json_kinds.F90(149): catastrophic error: Too many errors, exiting
compilation aborted for build\dependencies\json-fortran\src\json_kinds.F90 (code 1)
<ERROR> Compilation failed for object " build_dependencies_json-fortran_src_json_kinds.F90.o "
<ERROR>stopping due to failed compilation
STOP 1

Any clues ?

Thanks a lot

**Comment by milancurcic** [comment]
@jacobwilliams Has this come up before in json-fortran? Thanks!

**Reply by jacobwilliams** [reply]
I think that F90 files don't automatically get preprocessed on Windows. You have to add the fpp (or cpp? I can't remember) command line flag. 

I'm not at a pc right now. Try:

fpm build --flag "\fpp" 

Or something like that. 

Probably this should be added as a feature of FPM.

**Reply by jalvesz** [reply]
Yes, using --flag "/fpp ..." unblocked the issue! thanks a lot

... now I'm chasing all the "error stop _some_dynamic_string_" because intel19 doesn't allow for this kind of construct, it leads to: 
"error #6345: Either a PAUSE, STOP or ERROR STOP statement has an invalid argument." (it has to be constant) in neural-fortran there are just 3 files concerned but h5fortran is taking a while because this error stop was heavely used.

I had to come back to this kind of implementation:
from 
`if (io_status /= 0) error stop trim(io_message)`
to
`if (io_status /= 0) then`
`     write(*,*) trim(io_message)`
`     error stop` 
`end if`

**Reply by jacobwilliams** [reply]
Time to update your compiler. 😀
